tests/unit_tests/test_sky_import.py

```diff
@@ -0,0 +1,106 @@
+"""Ensure the lazy import of modules is not reverted.
+
+This test ensures that the lazy import of modules is not reverted.
+"""
+import sys
+
+import pytest
+
+
+def test_sky_import():
+    # Clean modules that are lazy imported by sky
+    modules_to_clean = [
+        module for module in list(sys.modules.keys()) if any(
+            module.startswith(prefix) for prefix in [
```

Comment:
Somewhat hard to maintain. IIUC, our intention is that a module being LazyImported should be LazyImported everywhere (otherwise lazy import just does not work). Can we just list all instances of `LazyImport` class and assert these modules are not actually imported? e.g.

```
objs = [obj for obj in gc.get_objects() if isinstance(obj, adaptor_common.LazyImport)]
```

Reply:
@Michaelvll What do you think? Is the above way good enough?